lib/api_callable.js

```diff
@@ -168,6 +168,12 @@ function addTimeoutArg(aFunc, timeout, otherArgs, abTests) {
     options.deadline = new Date(now.getTime() + timeout);
     var metadata =
       otherArgs.metadataBuilder ? otherArgs.metadataBuilder(abTests) : null;
+    var headers = otherArgs.headers || {};
+    for (var key in headers) {
+      if (headers.hasOwnProperty(key)) {
+        metadata.set(key, headers[key]);
```

Comment:
Ah good point. Maybe [`Metadata#add`](https://github.com/grpc/grpc/blob/master/src/node/src/metadata.js#L107) would work instead? This method simply appends the value if the key has not been set.